docs/documentation/coding-style.md

```diff
@@ -0,0 +1,49 @@
+# ILIAS Coding Style
+
+This is the coding style for the [ILIAS project](https://github.com/ILIAS-eLearning/ILIAS).
+
+## PSR-2 and additions
+
+The ILIAS coding standard is aligned to the widely used and established [PSR-2 standard](https://www.php-fig.org/psr/psr-2/)
+of the [PHP Interop Group (PHP-FIG)](https://www.php-fig.org/), extended by the
+following additions:
```

Comment:
Is is possible to add a reason why there are additions to PSR-2? Something like readability. I know code-style has a lot to do with personal taste, but there are also "real" reasons sometimes. Thanks!

Reply:
Yes, we can do that. But IMO they are not very interesting: They improve readability and PSR-2 does not address these issues =)

Keep in mind that PSR-2 was created based on commonalities among various projects and intentionally leaves out some areas. It thus establishes a minimum.

The spaces around colons for return-types are common in strictly typed languages (like [elm](https://elm-lang.org/docs/style-guide) or haskell) while in python the colon after a function head introduces the block with the statements for the function. Thus, for me the spaces around that colon feel just right. Maybe @legionth (in fact he proposed them) has more insights.

Reply:
> Yes, we can do that. But IMO they are not very interesting: They improve readability and PSR-2 does not address these issues =)
as I suspected. 👍 

> Maybe @legionth (in fact he proposed them) has more insights.
It is not necessary that this is discussed further. As I said, often it is just personal taste. I can live with almost everything :-) Main thing we have a defined code-style!

Reply:
Yes we can do this.
To list some facts:

Defined **with spaces** in:
* [Doctrine](https://www.doctrine-project.org/projects/doctrine-coding-standard/en/latest/reference/index.html#introduction)

Defined **without spaces**:
* https://github.com/php-fig/fig-standards/blob/master/proposed/extended-coding-style-guide.md
  * This is still in review and is still under discussion and additional proposals. So this could still change

**Not** defined but used **without spaces**:
* https://phpunit.de/getting-started/phpunit-8.html

Not defined at all:
* [Symfony](https://symfony.com/doc/master/contributing/code/standards.html) (to be fair the current version is still under development)

Doctrine is the first (it seems) which will establish such a rule and had always a big impact on other projects too.
The origin idea for this, was to make this rule similar to the operator rule which defines also spaces before and after the operator itself. 
TBH I have no strong feeling about this. To avoid discussion about the beauty of this, I would just go with the proposed rule of this PR.